radicle-tracker/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use std::hash;
+
+pub mod thread;
+use thread::*;
+
+pub struct Issue<Author> {
+    pub author: Author,
+    pub title: String,
+    pub thread: Thread<Comment<Author>>,
+    pub meta: (), // TODO(fintan): fill in meta data
+}
+
+impl<Author> Issue<Author> {
+    /// ```
+    /// use radicle_tracker::Issue;
+    ///
+    /// let issue = Issue::new(
+    ///     String::from("Monadic"),
+    ///     String::from("Buggy Boeuf"),
+    ///     String::from("We have bugs in our boeuf"),
+    /// );
+    ///
+    /// assert_eq!(issue.author, String::from("Monadic"));
+    /// assert_eq!(issue.title, String::from("Buggy Boeuf"));
+    /// assert_eq!(issue.thread.first().content, String::from("We have bugs in our boeuf"));
+    /// assert!(issue.thread.first().author == issue.author);
+    /// assert_eq!(issue.meta, ());
+    /// ```
+    pub fn new(author: Author, title: String, content: String) -> Self
+    where
+        Author: Clone + Eq,
+    {
+        let comment = Comment {
+            author: author.clone(),
+            content,
+        };
+
+        Issue {
+            author,
+            title,
+            thread: Thread::new(comment).0,
```

Comment:
It's a little unexpected that `Thread::new` doesn't return `Thread`

Reply:
Well it returns `Thread` and the `Path` to it. Does this break the idiom? It's useful to return the `Path` so that you can continue `reply`ing, `view`ing, or `delete`ing.

I can call it `create` instead to avoid confusion, whatya think?

Reply:
Yeah I think it does break the idiom. Can you not call `path()` to get the path?

Reply:
Or if you want to be fancy, you could `impl Into<Path> for Thread`

Reply:
But not sure if that makes sense, semantically :thinking: